core/src/SYCL/Kokkos_SYCL_Team.hpp

```diff
@@ -0,0 +1,633 @@
+  const auto team_rank = member.team_rank();
+  const auto nchunk    = (end - start + team_size - 1) / team_size;
+  value_type accum     = 0;
+  // each team has to process one or more chunks of the prefix scan
+  for (iType i = 0; i < nchunk; ++i) {
+    auto ii = start + i * team_size + team_rank;
+    // local accumulation for this chunk
+    value_type local_accum = 0;
+    // user updates value with prefix value
+    if (ii < loop_bounds.end) lambda(ii, local_accum, false);
+    // perform team scan
+    local_accum = member.team_scan(local_accum);
+    // add this blocks accum to total accumulation
+    auto val = accum + local_accum;
+    // user updates their data with total accumulation
+    if (ii < loop_bounds.end) lambda(ii, val, true);
+    // the last value needs to be propogated to next chunk
+    if (team_rank == team_size - 1) accum = val;
+    // broadcast last value to rest of the team
+    member.team_broadcast(accum, team_size - 1);
+  }
+}
+
+template <typename iType, class Closure>
+KOKKOS_INLINE_FUNCTION void parallel_for(
+    const Impl::TeamVectorRangeBoundariesStruct<iType, Impl::SYCLTeamMember>&
+        loop_boundaries,
+    const Closure& closure) {
+  loop_boundaries.member.group().parallel_for_work_item(
+      sycl::range<1>(loop_boundaries.end - loop_boundaries.start),
+      [=](sycl::h_item<1> item) {
+        closure(item.get_local()[0] + loop_boundaries.start);
+      });
+}
+
+template <typename iType, class Closure, class ReducerType>
+KOKKOS_INLINE_FUNCTION
+    typename std::enable_if<Kokkos::is_reducer<ReducerType>::value>::type
+    parallel_reduce(const Impl::TeamVectorRangeBoundariesStruct<
+                        iType, Impl::SYCLTeamMember>& /*loop_boundaries*/,
+                    const Closure& /*closure*/,
+                    const ReducerType& /*reducer*/) {
+  // FIXME_SYCL
+  Kokkos::abort("Not implemented!");
+}
+
+template <typename iType, class Closure, typename ValueType>
+KOKKOS_INLINE_FUNCTION
+    typename std::enable_if<!Kokkos::is_reducer<ValueType>::value>::type
+    parallel_reduce(const Impl::TeamVectorRangeBoundariesStruct<
+                        iType, Impl::SYCLTeamMember>& /*loop_boundaries*/,
+                    const Closure& /*closure*/, ValueType& /*result*/) {
+  // FIXME_SYCL
+  Kokkos::abort("Not implemented!");
+}
+
+//----------------------------------------------------------------------------
+
+/** \brief  Intra-thread vector parallel_for.
+ *
+ *  Executes closure(iType i) for each i=[0..N)
+ *
+ * The range [0..N) is mapped to all vector lanes of the calling thread.
+ */
+template <typename iType, class Closure>
+KOKKOS_INLINE_FUNCTION void parallel_for(
+    const Impl::ThreadVectorRangeBoundariesStruct<iType, Impl::SYCLTeamMember>&
+        loop_boundaries,
+    const Closure& closure) {
+  loop_boundaries.member.group().parallel_for_work_item(
+      sycl::range<1>(loop_boundaries.end - loop_boundaries.start),
```

Comment:
Uhm how does this behave if you are already inside a parallel for with TeamThreadRange? I.e. are you allowed to call a group().parallel_for_work_item inside a group().parallel_for_work_item where group is the same?

Reply:
Not sure. I replaced with a serial for-loop for now.